lib/components/mobile/batch-results-screen.js

```diff
@@ -40,30 +40,42 @@ const ExpandMapButton = styled(Button)`
   z-index: 999999;
 `
 
-const VIEW = uiActions.ItineraryView
-
 const NARRATIVE_SPLIT_TOP_PERCENT = 45
 
+// Styles for the results map also include prop-independent styles copied from mobile.css.
+const ResultsMap = styled.div`
+  bottom: ${props => props.expanded ? '0' : `${100 - NARRATIVE_SPLIT_TOP_PERCENT}%`};
+  display: ${props => props.visible ? 'inherit' : 'none'};
+  left: 0;
+  position: fixed;
+  right: 0;
+  top: 100px;
+`
+
+const StyledResultsError = styled(ResultsError)`
+  display: ${props => props.visible ? 'inherit' : 'none'};
+  top: ${props => props.visible ? (props.expanded ? '100px' : `${NARRATIVE_SPLIT_TOP_PERCENT}%`) : '100%'};
+  transition: top 300ms;
+`
+
+const VIEW = uiActions.ItineraryView
+
 /**
  * This component renders the mobile view of itinerary results from batch routing,
  * and features a split view between the map and itinerary results or narratives.
  */
 class BatchMobileResultsScreen extends Component {
-  componentDidUpdate (prevProps) {
-    if (this.props.activeLeg !== prevProps.activeLeg) {
-      // Check if the active leg has changed. If a different leg is selected,
-      // unexpand the itinerary to show the map focused on the selected leg
-      // (similar to the behavior of LineItinerary).
-      this._setItineraryExpanded(false)
-    }
+  state = {
+    previousItineraryView: null
   }
 
-  _setItineraryExpanded = itineraryExpanded => {
-    this.props.setItineraryView(itineraryExpanded ? VIEW.FULL : VIEW.SPLIT)
+  _setItineraryView = view => {
+    this.setState({ previousItineraryView: this.props.itineraryView })
+    this.props.setItineraryView(view)
   }
 
   _toggleMapExpanded = () => {
-    this.props.setItineraryView(this.props.itineraryView === VIEW.HIDDEN ? VIEW.SPLIT : VIEW.HIDDEN)
+    this._setItineraryView(this.props.itineraryView === VIEW.HIDDEN ? this.state.previousItineraryView : VIEW.HIDDEN)
```

Comment:
long line, please split up. Also, I think this could use a comment. What is the behavior here? The use of `previousItineraryView` is kind of confusing to me, especially given that the setState call happens within `_setItineraryView`.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/48bafb663209bf2e84d22b0c491709cba7c31919. We need the previous state to correctly display the itinerary as a list or the detailed itinerary when the user clicks `Show results`.